cumulus/steps/ec2/launch_config.py

```diff
@@ -0,0 +1,81 @@
+from troposphere import autoscaling, Ref, FindInMap, Base64, ec2
+from cumulus.chain import step
+import cumulus.components.userdata.linux
+from cumulus.steps.ec2 import META_SECURITY_GROUP_NAME
+
+
+class LaunchConfig(step.Step):
+
+    def __init__(self,
+                 asg_name,
+                 launch_config_name,
+                 meta_data,
+                 instance_profile_name,
+                 vpc_id=None,
+                 user_data=None, ):
+
+        step.Step.__init__(self)
+
+        self.asg_name = asg_name
+        self.launch_config_name = launch_config_name
+        self.user_data = user_data
+        self.meta_data = meta_data
+        self.instance_profile_name = instance_profile_name
+        self.vpc_id = vpc_id
+
+    def handle(self, chain_context):
+
+        template = chain_context.template
+
+        template.add_resource(ec2.SecurityGroup(
+            "ScalingGroupSecurityGroup",
+            GroupDescription="ScalingGroupSecurityGroup description",
+            **self._get_security_group_parameters()))
+
+        chain_context.metadata[META_SECURITY_GROUP_NAME] = "ScalingGroupSecurityGroup"
+
+        if not self.user_data:
+            user_data = cumulus.components.userdata.linux.user_data_for_cfn_init(
+                launch_config_name=self.launch_config_name,
+                asg_name=self.asg_name,
+                configsets='default',  # TODO: Fix this
+            )
+
+        launch_config = autoscaling.LaunchConfiguration(
+            self.launch_config_name,
+            UserData=Base64(user_data),
+            Metadata=self.meta_data,
+            IamInstanceProfile=Ref(self.instance_profile_name),
+            # search the template to find this instead of using name
+            **self._get_launch_configuration_parameters(chain_context)
+        )
+
+        template.add_resource(launch_config)
+
+    def _get_security_group_parameters(self):
+        config = {
+        }
+
+        if self.vpc_id:
+            config['VpcId'] = self.vpc_id
+
+        return config
+
+    def _get_launch_configuration_parameters(self, chain_context):
+
+        asg_sg_list = [self._get_launch_configuration_security_groups(chain_context.metadata[META_SECURITY_GROUP_NAME])]
+
+        parameters = {
+            'ImageId': FindInMap('AmiMap',
```

Comment:
ugh, so like stacker and parameters,  stacker has an opinion on mappings.  

These mappings are required to exist in stacker.yaml, for the sample code here.   This poses another problem if cumulus is going to be non-stacker specific..  

I think the best way to do this would be to supply a way to generate mappings in the way cumulus wants.  Something for a separate story/feature.